hermeto/core/resolver.py

```diff
@@ -64,14 +64,31 @@ def _resolve_packages(request: Request) -> RequestOutput:
     """Run all requested package managers, return their combined output."""
     _supported_package_managers = _package_managers
     requested_types = set(pkg.type for pkg in request.packages)
-    if "dev-package-managers" in request.flags:
-        _supported_package_managers = _package_managers | _dev_package_managers
+
+    # Convert x-<pkg> -> <pkg> and track which ones are experimental
+    # Use str instead of PackageManagerType since we're working with runtime values
+    normalized_types: set[str] = set()
+    experimental_types: set[str] = set()
+    for t in requested_types:
+        if not refers_to_experimental_pm(t):
+            normalized_types.add(t)
+        else:
+            # x-foo -> foo
+            normalized_type = t[2:]  # Remove "x-" prefix
```

Comment:
If you feel that explaining _what_ a bit of code does vs explaining why this has to be done it is a good candidate for a helper function. WDYT about this:
```python
def normalize_type(type_tag):
    # Currently experimental type tags are expected to start with "x-" prefix.
    # No regular type tag can have such prefix.
    return type_tag[2:]

... ...
   normalized_type = normalize_type(t)
```

Yes, this is a bit more verbose, but leaves no room for doubt or misinterpretation.